### ISSUE TITLE
Ability to issue and manage Gift Cards

### ISSUE THREAD

**Issue by rmehta** [frappe/erpnext#12789]
- [ ] Fields
  - [ ] Amount
  - [ ] Balance
  - [ ] Card Number
  - [ ] Allow only once
- [ ] Make Gift Card from Invoice

---

- [ ] Add "Gift Card" as Mode of Payment in Shopping Cart and POS

**Comment by palashjhabak** [comment]
Hi,

I want to do this feature but would need some guidance and discussions around it. My current questions are. 

1. Should we have gift cards ( apart from just creating and distributing ), as an option, linked to a customer because many companies issue cards to internal (employees) and external customers which they can refill and use to purchase anything. 

2. How do we handle the accounting entries of the gift card balance and refill of balances? Deferred Revenues as Current Liability? 

3. Should we then have a separate doctype for Gift Card Refill/Spend and a new doc created for each balance refill and spend just like its done in loyalty point entries. 

4. How about an option of converting the gift card balance to cash. 

Thanks,
Palash

**Comment by dealproc** [comment]
I'm making a shameless plug here... but... managing merchant liability is something that should be handled outside of the system, and have an integration into ERPNext.  

I've built a few merchant liability systems, and am working on a new one using EventStore as a backing system.  The advantage of using Event-Based storage is that you have a complete ledger of changes against the merchant liability.  If there is interest, I can provide APIs, etc. to build an integration.

To bring this full-circle, with ERPNext having OAuth capabilities, I am looking at using this feature with my ERPNext deployment, rather than building a custom system based on IdentityServer v4

In regard to liabilities, a historical report can be provided.  Once things are as they need to be, a data connector could be provided to bring the data into ERPNext from the merchant liability system and posted against the merchant's liability as-is necessary.

**Comment by deepeshgarg007** [comment]
Gift Cards can now be managed using coupon codes in ERPNext. This also has integration with POS checkout, closing this.
